File: 9/main.py
#!/usr/bin/env python


# integer sqrt
def isqrt(x):
	for i in range (x/2, 0, -1):
		if x == i * i:
			return i
	return -1


def main():
	for a in range(2, 500):
		for b in range(a-1, 2, -1):
			c = isqrt(a * a + b * b)
			if -1 == c:
				continue
		
			if 1000 == a + b + c:
				print '%d %d %d %d' % (a, b, c, a * b * c)
				return

main()

File: 9/main.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>


// integer sqrt
int isqrt(int x)
{
	for (int i = x/2; i > 0; i--)
	{
		if (x == i * i)
			return i;
	}
	return -1;
}


int main()
{
	for (int a = 2; a < 500; a++)
		{
		for (int b = a-1; b > 2; b--)
			{
			int c = isqrt(a * a + b * b);
			if (-1 == c)
				continue;
			
			if (1000 == a + b + c)
				{
				printf("%d\n", a * b * c);
				return 0;
				}
			}
		}
	return 1;
}
